src/libraries/Microsoft.Extensions.DependencyInjection.Specification.Tests/src/DependencyInjectionSpecificationTests.cs

```diff
@@ -761,6 +761,28 @@ public void ClosedServicesPreferredOverOpenGenericServices()
             Assert.IsType<FakeService>(service);
         }
 
+        // Reproduces https://github.com/dotnet/runtime/issues/79938
+        [Fact]
+        public void ResolvingEnumerableContainingOpenGenericServiceUsesCorrectSlot()
+        {
+            // Arrange
+            TestServiceCollection collection = new();
+            collection.AddTransient<IFakeOpenGenericService<PocoClass>, FakeService>();
+            collection.AddTransient(typeof(IFakeOpenGenericService<>), typeof(FakeOpenGenericService<>));
+            collection.AddSingleton<PocoClass>(); // needed for FakeOpenGenericService<>
+            IServiceProvider provider = CreateServiceProvider(collection);
+
+            // Act
+            IFakeOpenGenericService<PocoClass> service = provider.GetService<IFakeOpenGenericService<PocoClass>>();
+            IFakeOpenGenericService<PocoClass>[] services = provider.GetServices<IFakeOpenGenericService<PocoClass>>().ToArray();
+
+            // Assert
+            Assert.IsType<FakeService>(service);
+            Assert.Equal(2, services.Length);
+            Assert.True(services.Any(s => s.GetType() == typeof(FakeService)));
+            Assert.True(services.Any(s => s.GetType() == typeof(FakeOpenGenericService<PocoClass>)));
```

Comment:
Note: ideally we'd assert some order, but the Lamar container returns a different order than other implementations.